crates/ide/src/syntax_highlighting.rs

```diff
@@ -371,29 +373,20 @@ fn traverse(
         // string highlight injections, note this does not use the descended element as proc-macros
         // can rewrite string literals which invalidates our indices
         if let (Some(token), Some(descended_token)) = (token, descended_element.as_token()) {
-            let string = ast::String::cast(token);
-            let string_to_highlight = ast::String::cast(descended_token.clone());
-            if let Some((string, expanded_string)) = string.zip(string_to_highlight) {
+            if ast::String::can_cast(token.kind()) && ast::String::can_cast(descended_token.kind())
+            {
+                let string = ast::String::cast(token).unwrap();
+                let expanded_string = ast::String::cast(descended_token.clone()).unwrap();
```

Comment:
We can keep the pattern matching the `string.zip(string_to_highlight)` instead of unwrapping, likewise we can do a pattern match for the `ByteString` variants below. Unwrapping is usually a code smell so we want to avoid that if possible